lib/ansible/modules/system/sysctl.py

```diff
@@ -386,9 +386,21 @@ def main():
             ignoreerrors = dict(default=False, type='bool'),
             sysctl_file = dict(default='/etc/sysctl.conf', type='path')
         ),
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_if=[('state', 'present', ['value'])],
     )
 
+    if module.params['name'] is None:
+        module.fail_json(msg="name can not be None")
```

Comment:
Hmm, your first example should not return None values. If you want to assign None-values, you could do that using `null`. And that would be caught by the `required=True`.

So if this is true, that seems to be a bug.